### PR TITLE
Desktop: UI updates to sidebar and header, changing icon sizes and animations

### PR DESCRIPTION
**Pull request by innocuo**
- Added animation to icon in synchronize button
- Moved sync report above button, which prevents the sync button from moving from its place when the report has text.
- Added animation to icon in Toggle Sidebar button, using the css transition property.
- Reduced font size for text and icons in header and sidebar
- Changed theme color2 from white to a very light grey. It is barely noticeable, but reduces contrast a bit, improving readability.

### CODE REVIEW

**Comment by innocuo** [comment]
I posted in the forums a couple days ago proposing some design updates. [This is the related thread](https://discourse.joplinapp.org/t/some-design-suggestions-is-this-worth-implementing/2212).

Note that my PR is slightly different from what I posted in the screenshots in the forum:

- I didn't reduce the sidebar headers font size as much as in the screenshot.
- The Note List background color didn't changed (based on Laurent's feedback).
- This PR doesn't have the css that removes the gaps in the note list

Also, another thing I noticed: reducing the icon sizes makes it feel like the text is also smaller, but it isn't (the sidebar headers are slightly smaller, but the icon size reduction amplifies this). I tried to find a good average so the change is not too drastic.

**Comment by tessus** [comment]
> Added animation to icon in synchronize button

As soon as I click on the synchronize button, it switches to a cancel button, so how would an animated icon make sense? I certainly do not want that icon to turn over and over all the time.

File: ElectronClient/app/gui/Header.jsx [hunk 1]
```diff
@@ -102,11 +102,14 @@ class HeaderComponent extends React.Component {
 		let icon = null;
 		if (options.iconName) {
 			const iconStyle = {
-				fontSize: Math.round(style.fontSize * 1.4),
+				fontSize: Math.round(style.fontSize * 1.1),
 				color: style.color,
 			};
 			if (options.title) iconStyle.marginRight = 5;
-			if (options.iconRotation) iconStyle.transform = 'rotate(' + options.iconRotation + 'deg)';
+			if("undefined" != typeof(options.iconRotation)) {
```
Comment (laurent22) [inline ElectronClient/app/gui/Header.jsx]: Strict equality and no Yoda notation please.

File: ElectronClient/app/gui/SideBar.jsx [hunk 2]
```diff
@@ -642,9 +642,15 @@ class SideBarComponent extends React.Component {
 
 	synchronizeButton(type) {
 		const style = Object.assign({}, this.style().button, { marginBottom: 5 });
-		const iconName = type === "sync" ? "fa-refresh" : "fa-times";
+		const iconName = "fa-refresh";
 		const label = type === "sync" ? _("Synchronise") : _("Cancel");
-		const icon = <i style={{ fontSize: style.fontSize, marginRight: 5 }} className={"fa " + iconName} />;
+		let iconStyle = { fontSize: style.fontSize, marginRight: 5 };
```
Comment (laurent22) [inline ElectronClient/app/gui/SideBar.jsx]: `const` instead of `let`

**Comment by laurent22** [comment]
Thanks for the pull request. I've added a few comments to the code, and also if indeed you could clarify the behaviour of the sync button animation.

**Comment by innocuo** [comment]
Thanks for the comments, I'll make the requested changes.
I agree with you about yoda conditions, although I thought' "undefined"!=typeof whatever' can be consider an exception. 


My goal with all these changes were to do very small incremental changes that improve the UI but have little impact on the code (I'm an stranger to you and your code, so didn't want to waster your time).

Regarding the animated icon and label:

- I think visual feedback is useful to reassure users the app is "doing something". It's pretty common to have a progress bar or rotating circles animation when performing a network operation like refreshing, downloading or syncing data. Examples: instagram, twitter, dropbox, evernote, github desktop (the last 2 use similar rotating arrow icons).

- The sync report _is_ "visual feedback", but that's text in an app that is 90% text Little details in the interactions, like an animated icon, can add some life to the UI.

- I can't argue against users not wanting to see an animation, as that's user preference. I'd argue that in theory, users shouldn't be using the sync button often, as the app is supposed to handle synchronization automatically.

- A label of "synchronizing" would fit the rotating icon better, but then we need it translated 25 times. And then clicking on it to cancel would make less sense, so I discarded that.

- Keeping the icon, and start rotating it, gives continuity to the user action. And it's possitive feedback. Instead of telling them "you can cancel now", you're telling them "it's syncing now" (although it's 50/50 because the label still says cancel).


As a sidenote, these were other ideas I had about that feature, but completely out of scope:
- Cancelling a sync operation shouldn't be that standard. I think users should have the option, but it shouldn't be "that easy". A smaller x next to the sync report could be a better option than a big cancel button.
- the github app is a good example of combining both the sync button and the sync report in one, by updating the button text
- if you turn your wifi off, it seems the synchronization doesn't time out? Regardless, I think Electron can detect if you're online, so it can react to that. As examples, Toggl and Todoist are 2 apps that have offline warnings.
- the sync report should disappear after 1-2 minutes, not sure if it does?

**Comment by laurent22** [comment]
Thanks @innocuo, it looks good. After testing, I see that the animation is indeed only visible while synchronising, and I think that looks better than the previous cross icon.

**Pull request merged**